.github/workflows/cicd.yml

```diff
@@ -0,0 +1,186 @@
+name: CI/CD
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python 3.9
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.9
+
+    - name: Save pip cache
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/pip
+        key: ${{ runner.os }}-pip
+
+    - name: Upgrade pip
+      run: |
+        python -m pip install --upgrade --upgrade-strategy eager pip
+
+    - name: Install build dependencies
+      run: |
+        make deps-build
+
+    - name: Lint
+      run: make lint
+
+    - name: Static Analysis
+      run: make static
+
+    - name: Build
+      run: make build
+
+    - name: Archive build artifacts
```

Comment:
The reason I avoided upload is just because I did not want to give the impression that the artifacts are being uploaded for public consumption. I originally was just going to use the word "save", but thought "archive" sounded better. If it is confusing I don't mind changing it.